cou/apps/base.py

```diff
@@ -794,13 +794,28 @@ def _check_application_target(self, target: OpenStackRelease) -> None:
         if (
             self.current_os_release >= target
             and not self.can_upgrade_to
-            and self.apt_source_codename >= target
+            and not self._need_origin_setting_update(target)
         ):
             raise HaltUpgradePlanGeneration(
                 f"Application '{self.name}' already configured for release equal to or greater "
                 f"than {target}. Ignoring."
             )
 
+    def _need_origin_setting_update(self, target: OpenStackRelease) -> bool:
+        """Check if application needs to update origin setting for a certain target.
+
+        Applications with configuration equal or bigger than the target or without origin setting,
+        does not need to change origin setting.
```

Comment:
If this return False, we raised `HaltUpgradePlanGeneration`. My question is how operator will know that it's because of this check? e.g. self.apt_source_codename is None ?

Reply:
It's not because of this check, but because of the context. If the application is on an OpenStack release equal or bigger than the target, the channel doesn't have an available update and the origin setting is already as expected, there is nothing to do.

In the case of apps without origin settings we are always returning False, meaning that there is nothing to do with this configuration.

Reply:
But if these return `False` the `HaltUpgradePlanGeneration` will be raised with message saying that application is already configured release equal or greater than target. But if application simply do not have that configuration option we could provide another message or maybe even do another steps for upgrading it?